kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ScanSuite.scala

```diff
@@ -1594,6 +1594,51 @@ class ScanSuite extends AnyFunSuite with TestUtils with ExpressionTestUtils with
       )
     }
   }
+
+  Seq(
+    ("first checkpoint no predicate", None, Some(0), 2),
+    ("latest checkpoint no predicate", None, None, 4),
+    ("first checkpoint with predicate", Some(equals(col("id"), ofLong(10))), Some(0), 1)
```

Comment:
There is no version 10 in the table, how is this working?

Reply:
ah the "10" here is the predicate (i.e. `where id = 10`)